File: frontend/src/app/shipment/shipment-common/api/resources/invoice.resource.ts
export class InvoiceResource {


  constructor(trackingId: string, invoiceCreationDate: string,
              preCarriage: number, exportInsurance: number, exportCustomsClearance: number,
              flightPrice: number, importInsurance: number, importCustomsClearance: number,
              onCarriage: number, managementFee: number, serviceFee: number, discount: number) {
    this.trackingId = trackingId;
    this.invoiceCreationDate = invoiceCreationDate;
    this.preCarriage = preCarriage;
    this.exportInsurance = exportInsurance;
    this.exportCustomsClearance = exportCustomsClearance;
    this.flightPrice = flightPrice;
    this.importInsurance = importInsurance;
    this.importCustomsClearance = importCustomsClearance;
    this.onCarriage = onCarriage;
    this.managementFee = managementFee;
    this.serviceFee = serviceFee;
    this.discount = discount;
  }

  public trackingId: string;
  public invoiceCreationDate: string;
  public preCarriage: number;
  public exportInsurance: number;
  public exportCustomsClearance: number;
  public flightPrice: number;
  public importInsurance: number;
  public importCustomsClearance: number;
  public onCarriage: number;
  public managementFee: number;
  public serviceFee: number;
  public discount: number;
}

File: backend/src/main/java/org/educama/shipment/api/resource/InvoiceResource.java
package org.educama.shipment.api.resource;

import org.educama.shipment.model.Invoice;

/**
 * REST-Resource for single invoice.
 */
public class InvoiceResource {

    public String invoiceNumber;
    public String invoiceCreationDate;
    public double preCarriage;
    public double exportInsurance;
    public double exportCustomsClearance;
    public double flightPrice;
    public double importInsurance;
    public double importCustomsClearance;
    public double onCarriage;
    public double managementFee;
    public double serviceFee;
    public double discount;

    /**
     * Create a API-Model (Resource) instance from the internal data model.
     *
     * @param invoiceModel instance of the internal-data model
     * @return a converted InvoiceResource
     */

    public InvoiceResource fromInvoice(Invoice invoiceModel) {
        this.invoiceNumber = invoiceModel.invoiceNumber;
        this.invoiceCreationDate = invoiceModel.invoiceCreationDate.toString();
        this.preCarriage = invoiceModel.preCarriage;
        this.exportInsurance = invoiceModel.exportInsurance;
        this.exportCustomsClearance = invoiceModel.exportCustomsClearance;
        this.flightPrice = invoiceModel.flightPrice;
        this.importInsurance = invoiceModel.importInsurance;
        this.importCustomsClearance = invoiceModel.importCustomsClearance;
        this.onCarriage = invoiceModel.onCarriage;
        this.managementFee = invoiceModel.managementFee;
        this.serviceFee = invoiceModel.serviceFee;
        this.discount = invoiceModel.discount;
        return this;
    }
}
